Update and cleanup build

matrei opened the pull request:
- Remove buildSrc
- Remove pluginManagement
- Update dependencies
- Use Groovy provided by Gradle for documentation generation
- Set `grails-core` dependency as provided
- Remove redundant `assets` config
- Remove redundant Gradle plugins
- Use Groovy style in build files
- Set `grailsCentral` as the first dependency repository
- Move `sourceCompatibility` inside java config block (target is inherited)
- Add `textFixtures` sources to `groovydoc` generation
- Add `slf4j-nop` to clear compile warning
- Move all versions to `gradle.properties`
- Change to proper exclusions for `geb-spock` dependency
- Override `selenium` version set by `spring-boot-dependencies`
- Add `develocity` gradle plugins for build scans and remote cache

matrei committed:
fix(deps): update and cleanup build

- Remove buildSrc
- Remove pluginManagement
- Update dependencies
- Use Groovy provided by Gradle for documentation generation
- Set `grails-core` dependency as provided
- Remove redundant `assets` config
- Remove redundant Gradle plugins
- Use Groovy style in build files
- Set `grailsCentral` as the first dependency repository
- Move `sourceCompatibility` inside java config block (target is inherited)
- Add `textFixtures` sources to `groovydoc` generation
- Add `slf4j-nop` to clear compile warning
- Move all versions to `gradle.properties`
- Change to proper exclusions for `geb-spock` dependency
- Override `selenium` version set by `spring-boot-dependencies`

sbglasius reviewed (approved):
Very nice.

matrei committed:
ci: add develocity gradle plugins for build scans and remote cache

matrei committed:
ci: add build scan tags

jdaugherty commented on build.gradle:
```diff
@@ -20,34 +20,36 @@ apply plugin: 'org.grails.grails-plugin'
+    maven { url = 'https://repo.grails.org/grails/core' }
 }
 
 configurations {
```
Is this block doing anything now?

matrei replied:
Yes, it is registering a configuration named `documentation`. I'll rewrite it so it becomes more clear.

jdaugherty commented on build.gradle:
```diff
@@ -85,6 +88,10 @@ tasks.withType(Test).configureEach {
+    sourceCompatibility = JavaVersion.VERSION_11
+}
+
 tasks.named('bootJar') { enabled = false }
```
Is bootJar forced to false in the 6.x grails plugin gradle plugin?  Is this even needed?

matrei replied:
No, I'll remove it :+1:

jdaugherty commented on gradle.properties:
```diff
@@ -1,10 +1,14 @@
-seleniumVersion=4.1.4
-testcontainersVersion=1.20.2
+
+grailsVersion=6.2.2
```
This will mean we only support the latest grails 6.x in 4.1.x, right?  I would assume we want to support all of the 6.x line, specifically 6.1?

matrei replied:
No, I would not think so. Are there any breaking changes between `6.0.x` and `6.2.x`?

jdaugherty replied:
If we set the test app to 6.0.0 and the plugin to 6.2.2 we can show that?

jdaugherty commented on spock-container-test-app/build.gradle:
```diff
@@ -1,67 +1,66 @@
 }
 
-group = "org.demo.spock"
+apply plugin: 'groovy'
```
Why switch from the plugin{} syntax?

matrei replied:
```console
* What went wrong:
Plugin [id: 'org.grails.grails-web'] was not found in any of the following sources:

- Gradle Core Plugins (plugin is not in 'org.gradle' namespace)
- Plugin Repositories (plugin dependency must include a version number for this source)
```

jdaugherty replied:
Isn't that because you don't have something like this in the settings gradle:

```
pluginManagement {
    repositories {
        maven { url "https://repo.grails.org/grails/core/" }
        gradlePluginPortal()
    }
    plugins {
        id "org.grails.grails-web" version "${grailsGradlePluginVersion}"
        id "org.grails.grails-gsp" version "${grailsGradlePluginVersion}"
        id "org.grails.grails-plugin" version "${grailsGradlePluginVersion}"
        id "org.grails.grails-publish" version "${grailsGradlePluginVersion}"
    }
}
```

matrei replied:
Yes

jdaugherty commented on spock-container-test-app/build.gradle:
```diff
@@ -1,67 +1,66 @@
+        mavenCentral()
+    }
+    dependencies {
+        classpath "com.bertramlabs.plugins:asset-pipeline-gradle:$assetPipelineVersion"
```
Does this plugin really need the asset plugin?

matrei replied:
This is not the build file of the plugin, but of the functional tests project.

jdaugherty commented on spock-container-test-app/build.gradle:
```diff
@@ -1,67 +1,66 @@
 
-group = "org.demo.spock"
+apply plugin: 'groovy'
+apply plugin: 'org.grails.grails-web'
+apply plugin: 'org.grails.grails-gsp'
```
Should we be only applying the grails plugin gradle plugin?

matrei replied:
This build file is for the functional tests project, which is a grails app.

jdaugherty commented on spock-container-test-app/build.gradle:
```diff
@@ -1,67 +1,66 @@
-    testImplementation("org.spockframework:spock-core")
-    testImplementation("io.micronaut:micronaut-http-client")
+
+    implementation 'org.grails:grails-core'
+    implementation 'org.grails:grails-logging'
+    implementation 'org.grails:grails-plugin-databinding'
+    implementation 'org.grails:grails-plugin-i18n'
+    implementation 'org.grails:grails-plugin-interceptors'
+    implementation 'org.grails:grails-plugin-rest'
+    implementation 'org.grails:grails-plugin-services'
+    implementation 'org.grails:grails-plugin-url-mappings'
+    implementation 'org.grails:grails-web-boot'
+    implementation 'org.grails.plugins:gsp'
+    implementation 'org.grails.plugins:hibernate5'
+    implementation 'org.grails.plugins:scaffolding'
+    implementation 'org.springframework.boot:spring-boot-autoconfigure'
+    implementation 'org.springframework.boot:spring-boot-starter'
+    implementation 'org.springframework.boot:spring-boot-starter-actuator'
+    implementation 'org.springframework.boot:spring-boot-starter-logging'
+    implementation 'org.springframework.boot:spring-boot-starter-tomcat'
+    implementation 'org.springframework.boot:spring-boot-starter-validation'
+
+    compileOnly 'io.micronaut:micronaut-inject-groovy'
+
+    runtimeOnly "com.bertramlabs.plugins:asset-pipeline-grails:$assetPipelineVersion"
+    runtimeOnly 'com.h2database:h2'
+    runtimeOnly 'org.apache.tomcat:tomcat-jdbc'
+    runtimeOnly 'org.fusesource.jansi:jansi:1.18'
```
Since we really are only providing test classes, should this even add all of these grails dependencies?

matrei replied:
This is the test app.

jdaugherty commented on spock-container-test-app/gradle.properties:
```diff
@@ -1,6 +1,8 @@
-grailsGradlePluginVersion=6.0.0
 version=0.1
+
+grailsVersion=6.2.2
```
For the backport, I would choose the minimum grails version so people know the lowest level of grails we are compatible with

matrei replied:
This is the test app. The plugin should be compatible with all Grails 6 versions.

jdaugherty replied:
choosing the minimum version in the test app could show that 😀

matrei replied:
Yes, you could see it that way. In this case, until proven wrong, I chose to lean on semantic versioning rules :smile: 

Do you want me to change it back to `6.0.0`? (I have tested it with `6.0.0`, and it works)

jdaugherty replied:
If you have tested with it, I am good.  I hope we don't have that many more 6.x releases.

jdaugherty commented on spock-container-test-app/settings.gradle:
```diff
@@ -1,17 +1,31 @@
+develocity {
+    server = 'https://ge.grails.org'
+    buildScan {
+        tag('grails')
```
YES, thank you!

matrei committed:
test: set the correct `grails-gradle-plugin` version in the test app

matrei committed:
build: clarify and refactor documentation configuration registration

Refactored the registration logic for documentation configuration to enhance readability and make the purpose of the code clearer. This change improves maintainability and reduces potential confusion.

matrei committed:
chore: remove explicit `bootJar` task disabling

The grails-gradle-plugin now handles this functionality, so the explicit disabling is no longer necessary.

matrei committed:
chore: cleanup
